store/src/java/com/zimbra/cs/service/mail/SaveProfileImage.java

```diff
@@ -73,41 +81,14 @@ public Element handle(Element request, Map<String, Object> context) throws Servi
                 throw MailServiceException
                     .INVALID_IMAGE("Uploaded image is not a valid image file");
             }
-            try {
-                Folder imgFolder = mbox.getFolderByName(octxt, Mailbox.ID_FOLDER_ROOT, folderName);
-                folderId = imgFolder.getId();
-            } catch (ServiceException exception) {
-                if (MailServiceException.NO_SUCH_FOLDER.equals(exception.getCode())) {
-                    byte hidden = Folder.FOLDER_IS_IMMUTABLE | Folder.FOLDER_DONT_TRACK_COUNTS;
-                    Folder.FolderOptions fopt = new Folder.FolderOptions();
-                    fopt.setAttributes(hidden).setDefaultView(MailItem.Type.DOCUMENT).setFlags(0)
-                        .setColor(MailItem.DEFAULT_COLOR_RGB);
-                    Folder profileFolder = mbox.createFolder(octxt, folderName,
-                        Mailbox.ID_FOLDER_ROOT, fopt);
-                    folderId = profileFolder.getId();
-                } else {
-                    throw ServiceException.FAILURE(FAILURE_MESSAGE, exception);
-                }
-            }
-            Doc imageDoc = new Doc(in, new ContentType(contentType, "image/jpeg"), name, null,
-                "Account Profile Image");
-            try {
-                TypedIdList ids = mbox.getItemIds(octxt, folderId);
-                List<Integer> idList = ids.getAllIds();
-                MailItem[] itemList = mbox.getItemById(octxt, idList, MailItem.Type.DOCUMENT);
-                for (MailItem item : itemList) {
-                    CustomMetadata customData = item.getCustomData(IMAGE_CUSTOM_DATA_SECTION);
-                    if (customData != null && customData.containsKey("p")
-                        && customData.get("p").equals("1")) {
-                        mbox.delete(octxt, item.getId(), MailItem.Type.DOCUMENT);
-                    }
-                }
-                imgDocItem = createDocument(imageDoc, zsc, octxt, mbox, null, in, folderId,
-                    MailItem.Type.DOCUMENT, null,
-                    new CustomMetadata(IMAGE_CUSTOM_DATA_SECTION, "d1:p1:1e"), false);
-            } catch (ServiceException e) {
-                throw ServiceException.FAILURE(FAILURE_MESSAGE, e);
+            if (inputSize > 2097152l) {
```

Comment:
Can you please add 2097152l in debug config? 
2097152l seems bit odd number.  
can we have something more intuitive like 2Mb.

Reply:
Done. changed it to 2*1024*1024.